pkg/controller/highperf/highperf_controller.go

```diff
@@ -0,0 +1,234 @@
+		return
+	}
+
+	if !cache.WaitForNamedCacheSync("networks", stopCh, c.nwInformer.HasSynced) {
+		return
+	}
+
+	if !cache.WaitForNamedCacheSync("gkenetworkparamset", stopCh, c.gnpInformer.HasSynced) {
+		return
+	}
+
+	for i := 0; i < numWorkers; i++ {
+		go wait.UntilWithContext(ctx, c.runWorker, time.Second)
+	}
+
+	<-stopCh
+}
+
+func (c *Controller) runWorker(ctx context.Context) {
+	for c.processNextItem(ctx) {
+	}
+}
+
+func (c *Controller) processNextItem(ctx context.Context) bool {
+	key, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+
+	defer c.queue.Done(key)
+
+	c.processNode(ctx, key.(string))
+	// TODO (bmorettodama): handle errors from processNode (try again)
+	c.queue.Forget(key)
+	return true
+}
+
+func (c *Controller) processNode(ctx context.Context, key string) error {
+	node, err := c.nodeInformer.Lister().Get(key)
+
+	if err != nil {
+		klog.Error("Fetching object with key %s from store failed with %v", key, err)
+		return err
+	}
+
+	// check if northInterface and network-status have entries for same network
+	northInterfaces, err := networkv1.ParseNorthInterfacesAnnotation(node.Annotations[networkv1.NorthInterfacesAnnotationKey])
+	if err != nil {
+		klog.Errorf("Parsing northInterface annotation: %v", err)
+		return err
+	}
+	nodeNetworks, err := networkv1.ParseNodeNetworkAnnotation(node.Annotations[networkv1.NodeNetworkAnnotationKey])
+	if err != nil {
+		klog.Errorf("Parsing network-status annotation: %v", err)
+		return err
+	}
+	var readyNetworks networkv1.NorthInterfacesAnnotation
+
+	// create a list of networks present in both the northInterface and network-status annotations
+	statusMap := make(map[string]string)
+	for _, nodeNetwork := range nodeNetworks {
+		statusMap[nodeNetwork.Name] = ""
+	}
+	for _, northInterface := range northInterfaces {
+		if _, ok := statusMap[northInterface.Network]; ok {
+			readyNetworks = append(readyNetworks, northInterface)
+		}
+	}
+
+	// return when there are no ready netwoks
+	if len(readyNetworks) == 0 {
+		return nil
+	}
+
+	for _, network := range readyNetworks {
+		// Fetch network object and check if its Device type
+		obj, exists, err := c.nwInformer.GetIndexer().GetByKey(network.Network)
+		if err != nil {
+			klog.Errorf("Fetching %s network object: %v", network.Network, err)
+			return err
+		}
+		if !exists {
+			return nil
+		}
+		nwobj := obj.(*networkv1.Network)
+
+		if nwobj.Spec.Type != networkv1.DeviceNetworkType {
+			continue
+		}
+
+		klog.Info("Network %s is of type Device", network.Network)
+		obj, exists = nil, false
+
+		// Fetch GNP object for Device type networks
+		obj, exists, err = c.gnpInformer.GetIndexer().GetByKey(nwobj.Spec.ParametersRef.Name)
+		if err != nil {
+			klog.Error("Fetching %s gnp object: %v", nwobj.Spec.ParametersRef.Name, err)
+			return err
+		}
+		if !exists {
+			return nil
+		}
+		gnpobj := obj.(*networkv1alpha1.GKENetworkParamSet)
+
+		c.updateConfigMap(node, nwobj, gnpobj)
+		c.updateNodeStateSpec(node, nwobj, gnpobj)
```

Comment:
check the errors

Reply:
is better to add it, I wish we have a linter to flag these things, this code goes to production, we should treat the code as is, despite we know is not going to run by default. Who knows in the future if anybody else has to pick the task, that person will not have this context, 
```suggestion
		err = c.updateConfigMap(node, nwobj, gnpobj)
		if err != nil {
		  return err
		 }
		c.updateNodeStateSpec(node, nwobj, gnpobj)
		if err != nil {
		  return err
		 }
```

Reply:
Sure